modules/nextflow/src/main/groovy/nextflow/processor/TaskRun.groovy

```diff
@@ -372,23 +372,31 @@ class TaskRun implements Cloneable {
         return processor.singleton ? processor.name : "$processor.name ($index)"
     }
 
+    String getTag() {
+        if (config.containsKey('tag')) {
+            try {
+                // -- look-up the 'sampleId' property, and if everything is fine
+                //    cache this value in the 'name' attribute
+                return String.valueOf(config.tag).trim()
```

Comment:
this will return `"null"` if `config.tag == null` - is this desired behavior?

Reply:
feel like it makes more sense to return `null` if `config.tag` is `null`

Reply:
good catch